thoth/storages/graph/dgraph.py

```diff
@@ -1299,21 +1299,23 @@ def _python_sync_analysis_result(
                 # setup.py that is not distutils setup.py
                 _LOGGER.info("Skipping error entry - %r", python_package_info)
                 continue
-
-            python_package_version = PythonPackageVersion.from_properties(
-                ecosystem="python",
-                package_name=python_package_info["result"]["name"],
-                package_version=python_package_info["result"]["version"],
-                index_url=None,
-                extras=None,
-                os_name=environment.os_name,
-                os_version=environment.os_version,
-                python_version=environment.python_version,
-                solver_error=False,
-                solver_error_unparseable=False,
-                solver_error_unsolvable=False,
-            )
-            self._create_python_package_record(python_package_version, verify_index=False)
+            try:
+                python_package_version = PythonPackageVersion.from_properties(
+                    ecosystem="python",
+                    package_name=python_package_info["result"]["name"],
+                    package_version=python_package_info["result"]["version"],
+                    index_url=None,
+                    extras=None,
+                    os_name=environment.os_name,
+                    os_version=environment.os_version,
+                    python_version=environment.python_version,
+                    solver_error=False,
+                    solver_error_unparseable=False,
+                    solver_error_unsolvable=False,
+                )
+                self._create_python_package_record(python_package_version, verify_index=False)
+            except Exception:  # pylint: disable=broad-exception
+               _LOGGER.exception(f"Failed to sync Python package, error is not fatal: {python_package_info!r}")
```

Comment:
This will cause exceptions being reported when we detect a setup.py in a container image which does not correspond to any package (I saw these happening).

Instead, it would be better to add an if guard before creating `PythonPackageVersion`:

```python
if not python_package_info["result"].get("name"):
    _LOGGER.warning("No package name found in entry %r when syncing document %r", python_package_info, analysis_document_id)
    continue
```

and then follow with python package model creation.